### PR TITLE
feat[WIP]: Videos

### PR DESCRIPTION
**Pull request by saadahmadhere**
This PR consists of adding videos to the database, fetching and displaying them on the video listing page with just the basic(static) functionality.

### CODE REVIEW

**Review by Hamdrive — COMMENTED** [review]
Liked the implementation of custom hook to detect user scroll for opened menu
overall good structuring done

File: src/Helper/Hooks/useClickOutside.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { useEffect, useRef } from "react";
+
+const useClickOutside = (handler) => {
```
Comment (hinanshisuthar) [inline src/Helper/Hooks/useClickOutside.js:3]: Nice use of useRef here

File: src/Helper/formatViewsCount.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const formatViewsCount = (views) => {
+  if (views > 999 && views < 1000000) {
```
Comment (hinanshisuthar) [inline src/Helper/formatViewsCount.js:2]: Nice how you have calculated this

File: src/components/Video-Card/Card.jsx [hunk 3]
```diff
@@ -1,41 +1,69 @@
...
+                  : video.creator}
+              </p>
               <div className="video_details mt_1">
-                <span className="video_views">999 views</span>
-                <span className="video_upload_date">11 months ago</span>
+                <span className="video_views">{`${formatViewsCount(
+                  video.views
+                )} views`}</span>
+                <span className="video_upload_date">
+                  {new Date(video.uploaded).toDateString().slice(4)}
```
Comment (hinanshisuthar) [inline src/components/Video-Card/Card.jsx:42]: You are already using new Date in the videos db. Hence, no need here

**Review by hinanshisuthar — COMMENTED** [review]
Overall implementation looks good. Nice and clean UI. Good to merge.

File: src/components/Video-Card/Card.jsx (hunk 3 above)
Comment (saadahmadhere) [reply src/components/Video-Card/Card.jsx:42]: Yes, but the new Date from the videos DB sends a string that needs to be converted back to the date object to perform the desired operations.

**Pull request merged**